src/components/CreateTemplate/index.tsx

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable require-jsdoc */
+import { useStoreSelector } from "@/store/hooks";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  FormControl,
+  FormLabel,
+  Input,
+  Box,
+  Flex,
+  Text,
+  Textarea,
+  Icon,
+  Badge,
+  useDisclosure,
+} from "@chakra-ui/react";
+import React, { useRef, useState } from "react";
+import { FaUserCircle } from "react-icons/fa";
+import { useAddTemplateMutation } from "@/redux/api/templateApi";
+import SuccessNotice from "./SuccessNotice";
+import generateNewTemplate from "@/utils/generateNewTemplate";
+import { userNameEllip } from "../../utils/handleLongUserName";
+
+interface Props {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const maxCourtNameLen = 15;
+const maxDescriptionLen = 200;
+const maxUserNameDisplay = 20;
+const regex = /[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g;
+
+const inputErrorInit = {
+  courtNameFullErr: false,
+  courtNameNullErr: false,
+  descriptionOverLimit: false,
+};
+
+const inputErrorMsg = {
+  nameFullErrMsg: `Court name cannot have more than ${maxCourtNameLen} characters`,
+  nameNullErrMsg: "Court name cannot be empty",
+  descriptionLenErrMsg: `Description can not over ${maxDescriptionLen} words`,
+};
+
+function CreateTemplate({ isOpen, onClose }: Props) {
+  const courtNameRef = useRef<HTMLInputElement>(null);
+  const descriptionRef = useRef<HTMLTextAreaElement>(null);
+  const [inputError, setInputError] = useState(inputErrorInit);
+  const [textAreaLen, setTextAreaLen] = useState(0);
+  const { userId, firstName, lastName } = useStoreSelector((state) => state.user);
+  const { activeCourt: selectedCourt } = useStoreSelector((state) => state.courtSpecData);
+  const { court: selectedCourtTileData } = useStoreSelector((state) => state.tile.present);
+  const [addTemplate] = useAddTemplateMutation();
+  const courtType = "basketball";
+  const userFullName = userNameEllip(`${firstName} ${lastName}`, maxUserNameDisplay);
+  const {
+    isOpen: isSuccessNoticeOpen,
+    onOpen: handleSuccessNoticeOpen,
+    onClose: handleSuccessNoticeClose,
+  } = useDisclosure();
+
+  const closeWindow = () => {
+    setInputError(inputErrorInit);
+    setTextAreaLen(0);
+    onClose();
+  };
+
+  const checkNameLength = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const nameInputLen = e.currentTarget.value.length;
+    setInputError((inputError) => ({
+      ...inputError,
+      courtNameFullErr: nameInputLen > maxCourtNameLen,
+      courtNameNullErr: nameInputLen === 0,
+    }));
+  };
+
+  const handleTextAreaLenChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const textAreaLength = e.currentTarget.value
+      .replace(regex, " ")
+      .split(" ")
+      .filter((item) => item != "").length;
+    setTextAreaLen(textAreaLength);
+    setInputError((inputError) => ({
+      ...inputError,
+      descriptionOverLimit: textAreaLength > maxDescriptionLen,
+    }));
+  };
```

Comment:
If the special symbols are not counted here, then if I enter 201 @, it will not count as exceeding the word count requirement?

Reply:
Yes, we only count the normal punctuations which we can directly input from keyboard. If user type some special symbols other than our regex, they will be counted as normal characters.

Reply:
> If the special symbols are not counted here, then if I enter 201 @, it will not count as exceeding the word count requirement?

Yes, we only count the normal punctuations which we can directly input from keyboard. If user type some special symbols other than our regex, they will be counted as normal characters.